### DISCUSSION TITLE
Is there a way to extract the Elevation data from the valhalla tiles

### DISCUSSION THREAD

**Discussion by abidbodal** [valhalla/valhalla#3174] [Q&A] [answered]
I am trying to create a route with an elevation profile, so I am looking for an API where I can get a payload back with the elevation profile along a route. Alternatively is there an API where I can query valhalla for the elevation, given the lat, lon?

**Comment by kevinkreiser** [comment, marked as answer]
its not in the graph tiles (yet) but you can do exactly that with the elevation tiles. you must download the elevation tiles using `valhalla_build_elevation` and then run the valhalla services as normal and call the `/height` service. please read the documentation https://github.com/valhalla/valhalla/blob/master/docs/api/elevation/api-reference.md

**Reply by abidbodal** [reply]
Thanks
